core/src/banking_stage/consume_worker.rs

```diff
@@ -784,6 +780,55 @@ impl ConsumeWorkerTransactionErrorMetrics {
     }
 }
 
+#[derive(Debug)]
+pub enum TryRecvError {
+    Empty,
+    Disconnected,
+}
+
+#[derive(Debug)]
+pub enum TrySendError {
+    Full,
+    Disconnected,
+}
+
+pub(crate) trait ConsumeWorkerChannels {
+    type Tx: TransactionWithMeta;
+    fn try_recv(&mut self) -> Result<ConsumeWork<Self::Tx>, TryRecvError>;
+    fn try_send(
+        &mut self,
```

Comment:
note: for crossbeam these are not necessary to be `mut`. However for `shaq` we update some local usizes internally.

Likely when we get to point of using shaq we may need to add something like `sync/finalize`. Don't need those with crossbeam either, but trying to keep this PR simpler. We don't want to call them every time we recv/send for efficiency.

Reply:
preferably keep try_recv and try_send without `mut`, so interface close to standard; sounds like it's being `mut` is due to some "internal" logic that doesn't have to export to callers. If it is as simple as replace uszies with atomic usizes, maybe it worth change; otherwise can look into it separately from this PR.